internal/controller/progressive/progressive.go

```diff
@@ -246,110 +253,85 @@ func GetChildName(ctx context.Context, rolloutObject ctlrcommon.RolloutObject, c
 	}
 }
 
-// return whether we're done, and error if any
+// return:
+// - whether we're done
+// - whether we just created a new child
+// - error if any
 func processUpgradingChild(
 	ctx context.Context,
 	rolloutObject ctlrcommon.RolloutObject,
 	controller progressiveController,
 	existingPromotedChildDef, existingUpgradingChildDef *unstructured.Unstructured,
 	c client.Client,
-) (bool, error) {
+) (bool, bool, error) {
 	numaLogger := logger.FromContext(ctx)
 
-	assessment, err := assessUpgradingChild(ctx, existingUpgradingChildDef)
+	assessment, err := controller.AssessUpgradingChild(ctx, existingUpgradingChildDef)
 	if err != nil {
-		return false, err
+		return false, false, err
 	}
 	numaLogger.WithValues("name", existingUpgradingChildDef.GetName()).Debugf("assessment returned: %v", assessment)
 
 	switch assessment {
-	case AssessmentResultFailure:
+	case apiv1.AssessmentResultFailure:
 
 		rolloutObject.GetRolloutStatus().MarkProgressiveUpgradeFailed(fmt.Sprintf("New Child Object %s/%s Failed", existingUpgradingChildDef.GetNamespace(), existingUpgradingChildDef.GetName()), rolloutObject.GetRolloutObjectMeta().Generation)
+		rolloutObject.GetRolloutStatus().ProgressiveStatus.UpgradingChildStatus = &apiv1.ChildStatus{Name: existingUpgradingChildDef.GetName(), AssessmentResult: apiv1.AssessmentResultFailure}
```

Comment:
this information from the PipelineRollout is what informs the ISBServiceRollout reconciler as to what happened.